### PR TITLE
feat: add support for multi-month display to DatePicker

### PR DESCRIPTION
**Pull request by mkrl**
[FX-3963](https://toptal-core.atlassian.net/browse/FX-3963)

### Description

Add support for rendering two months at the same time to DatePicker.
This is only available on screens > `sm` range.

### How to test

- Take a look at the [deployed storybook](https://picasso.toptal.net/fx-3963-support-two-month-calendar-visualization-in-the-datepicker-component/?path=/story/forms-datepicker--datepicker#with-two-months)
- Check if the code looks good
- Check if combinations of other props work well with two month view.

### Screenshots

| Before.                                 | After.                                  |
| --------------------------------------- | --------------------------------------- |
| image | image |

### Development checks

- [n/a] Add changeset according to [guidelines](https://github.com/toptal/picasso/blob/master/docs/contribution/changeset-guidelines.md) (if needed)
- [x] Read [CONTRIBUTING.md](https://github.com/toptal/picasso/blob/master/CONTRIBUTING.md) and [Component API principles](https://github.com/toptal/picasso/blob/master/docs/contribution/component-api.md)
- [x] Make sure that additions and changes on design follow [Toptal's BASE design](https://design.toptal.net/), and it's been already discussed with designers at #-base-core
- [x] Annotate all `props` in component with documentation
- [x] Create `examples` for component
- [x] Ensure that deployed demo has expected results and good examples
- [x] Ensure the changed/created components have not caused accessibility issues. [How to use accessibility plugin in storybook](https://github.com/toptal/picasso/blob/master/docs/contribution/accessibility.md).
- [x] Self reviewed
- [x] Covered with tests

> All **_development checks_** should be done and set checked to pass the
> **GitHub Bot: TODOLess** action

<details>
<summary>PR commands</summary>
<br />

List of available commands:

- `@toptal-bot run package:alpha-release` - Release alpha version
- `@toptal-anvil ping reviewers` - Ping FX team for review

</details>

<details>
<summary>PR Review Guidelines</summary>
<br />

#### When to approve? ✅

**You are OK** with merging this PR and

1. You have no extra requests.
2. You have optional requests.
   1. Add `nit:` to your comment. (ex. `nit: I'd rename this variable from makeCircle to getCircle`)

#### When to request changes? ❌

**You are not OK** with merging this PR because

1. Something is broken after the changes.
2. Acceptance criteria is not reached.
3. Code is dirty.

#### When to comment (neither ✅ nor ❌)

**You want your comments to be addressed** before merging this PR in cases like:

1. There are leftovers like unnecessary logs, comments, etc.
2. You have an opinionated comment regarding the code that requires a discussion.
3. You have questions.

#### How to handle the comments?

1. An owner of a comment is the only one who can resolve it.
2. An owner of a comment must resolve it when it's addressed.
3. A PR owner must reply with ✅ when a comment is addressed.

</details>

[FX-3963]: https://toptal-core.atlassian.net/browse/FX-3963?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/picasso/src/CalendarMonthHeader/CalendarMonthHeader.tsx [hunk 1]
```diff
@@ -82,7 +82,9 @@ const CalendarMonthHeader = (props: CaptionProps) => {
 
   const defaultMarkup = (
     <Container flex justifyContent='space-between' bottom='medium'>
-      {!hidePrevious && (
+      {hidePrevious ? (
+        <div />
+      ) : (
```
Comment (mkrl) [inline packages/picasso/src/CalendarMonthHeader/CalendarMonthHeader.tsx:87]: Those empty `div`s allow for proper months navigation alignment in the header. The design implies to have the same alignment as in single month view, but with one of the navigation buttons hidden.

image

**Comment by mkrl** [comment]
@toptal-anvil ping reviewers

File: packages/picasso/src/Calendar/Calendar.tsx [hunk 2]
```diff
@@ -49,6 +50,7 @@ export interface Props
   indicatedIntervals?: CalendarDateRange[]
   weekStartsOn?: WeekStart
   hasFooter?: boolean
+  showTwoMonths?: boolean
```
[the remark this answers is not in the document]
Comment (mkrl) [reply packages/picasso/src/Calendar/Calendar.tsx:53]: That was originally the plan, but I assume based on the requirements in the ticket that we want that to be limited to only one or two.
But limiting it with a type is also a good idea, thanks!

**Comment by dmaklygin** [comment]
@toptal-anvil ping reviewers

File: packages/picasso/src/Calendar/Calendar.tsx [hunk 3]
```diff
@@ -173,7 +181,10 @@ export const Calendar = forwardRef<HTMLDivElement, Props>(function Calendar(
           renderMonthHeader,
         }}
       >
-        <CalendarContainer hasFooter={hasFooter}>
+        <CalendarContainer
+          hasFooter={hasFooter}
+          showTwoMonths={shouldRenderMultipleMonths}
```
Comment (dmaklygin) [inline packages/picasso/src/Calendar/Calendar.tsx:186]: I still think, we also should use here `numberOfMonths`.

File: packages/picasso/src/CalendarContainer/CalendarContainer.tsx [hunk 4]
```diff
@@ -28,6 +33,7 @@ const CalendarContainer = ({ children, hasFooter }: CalendarContainerProps) => {
     <div
       className={cx(classes.root, {
         [classes.hasFooter]: hasFooter,
+        [classes.wide]: showTwoMonths,
```
Comment (dmaklygin) [inline packages/picasso/src/CalendarContainer/CalendarContainer.tsx:36]: I would specify screen size for each appearance: for single month and for two month appearances

Comment (mkrl) [reply packages/picasso/src/CalendarContainer/CalendarContainer.tsx]: I have a better idea. We don't need the fixed width with multiple months, I've updated it with a `flexible` prop so the width is automatic depending on the content. 
This way we can have any amount of months we want (with just extending the type signature).

**Comment by sashuk** [comment]
Should the next month button be blocked in case `maxDate` is set (same for the previous month and `minDate`)?


https://github.com/toptal/picasso/assets/1390758/ab0a356e-8228-480a-ae10-8854a5166954

**Review by dmaklygin — APPROVED** [review]
Good work!

**Comment by mkrl** [comment]
@sashuk I think it works as expected. Month navigation buttons does not allow for going beyond selectable limits.
Just realized it is different with the previous calendar implementation, but that is true for default configuration (one month) as well.

I could not find an option to disable it though. It's not related to this PR, but we can think this over in the PR to merge the new calendar into master.

**Review by sashuk — APPROVED** [review]
Code looks good!

**Pull request merged**